### ISSUE TITLE
iOS BoringSSL output frames and disconnects

### ISSUE THREAD

**Issue by rolinger** [dpa99c/cordova-plugin-firebasex#90]
# Bug report

**Current behavior:**
About every 4th or 5th build the app on the physical device hangs during the launch and the below message appears in Xcode console:

```
[BoringSSL] nw_protocol_boringssl_input_finished(1543) [C3.1:2][0x12fd335c0] Peer disconnected during the middle of a handshake. Sending errSSLClosedNoNotify(-9816) alert
TIC TCP Conn Failed [3:0x2805712c0]: 3:-9816 Err(-9816)
[BoringSSL] nw_protocol_boringssl_input_finished(1543) [C5.1:2][0x12fe46470] Peer disconnected during the middle of a handshake. Sending errSSLClosedNoNotify(-9816) alert
TIC TCP Conn Failed [5:0x280575f80]: 3:-9816 Err(-9816)
[BoringSSL] boringssl_context_alert_callback_handler(3724) [C6.1:2][0x12fd43710] Alert level: fatal, description: inappropriate fallback
[BoringSSL] boringssl_session_errorlog(224) [C6.1:2][0x12fd43710] [boringssl_session_handshake_incomplete] SSL_ERROR_SSL(1): operation failed within the library
[BoringSSL] boringssl_session_handshake_error_print(205) [C6.1:2][0x12fd43710] 5097281768:error:1000043e:SSL routines:OPENSSL_internal:TLSV1_ALERT_INAPPROPRIATE_FALLBACK:/BuildRoot/Library/Caches/com.apple.xbs/Sources/boringssl/boringssl-109.250.2/ssl/tls_record.cc:586:SSL alert number 86
[BoringSSL] boringssl_context_get_error_code(3617) [C6.1:2][0x12fd43710] SSL_AD_INAPPROPRIATE_FALLBACK
TIC TCP Conn Failed [6:0x280576640]: 3:-9860 Err(-9860)
NSURLSession/NSURLConnection HTTP load failed (kCFStreamErrorDomainSSL, -9860)
Task <02ADA890-35C5-4DE7-B0E5-6EC812CF79E6>.<1> HTTP load failed (error code: -1200 [3:-9860])
Task <02ADA890-35C5-4DE7-B0E5-6EC812CF79E6>.<1> finished with error - code: -1200
nw_protocol_boringssl_get_output_frames(1301) [C1.1:2][0x12fe15ea0] get output frames failed, state 8196
nw_protocol_boringssl_get_output_frames(1301) [C1.1:2][0x12fe15ea0] get output frames failed, state 8196
```

In addition, In Xcode, on every build to a physical device getting repeating messages I have never seen before, the below message always shows in pairs:

> [BoringSSL] nw_protocol_boringssl_get_output_frames(1301) [C7.1:2][0x129e062b0] get output frames failed, state 8196

Upon building the app to a physical device. 

This never happened before on previous versions of `Cordova-plugin-firebase` or even `Cordova-plugin-firebase-lib` - at least it was never observed.  When moved to `firebase` this started happening....no other plugins were added or updated.  The only code changes made were to deal with `breaking changes` between firebase and firebase

Upon researching this BoringSSL issue, came across a few threads at various sites refer to this as a Firebase issue.  In addition, the BoringSSL messages/disconnects happen immediately after Firebase loads. 

IE: https://github.com/firebase/firebase-ios-sdk/issues/357
**Environment information**
system information:

Cordova CLI: 9.0.0 (cordova-lib@9.0.1)
cordova-ios: 5.0.0
Gulp version:  CLI version 3.9.1
Gulp local:  
Ionic Framework Version: 1.3.4
Ionic CLI Version: 1.7.16
Ionic App Lib Version: 0.7.3
ios-deploy version: 1.9.4 
ios-sim version: 8.0.1 
OS: Mac OS X El Capitan
Node Version: v6.11.4
Xcode version: Xcode 10.1 Build version 10B61

**Comment by dpa99c** [comment]
I don't believe that this plugin is directly responsible for this or can do anything about it since it's simply a thin wrapper around the Firebase SDK and as [the linked issue](https://github.com/firebase/firebase-ios-sdk/issues/357) suggests, this is also arising in native iOS apps which access the Firebase SDK directly.
Therefore I'm closing this as I don't believe there's anything that the plugin can do to resolve it.

**Comment by rolinger** [comment]
@dpa99c - ok, thanks for taking a look at it.

**Comment by orianweste** [comment]
Hey @rolinger,
do you have anything new about this? Iam wasting hour after hour on this....

**Comment by rolinger** [comment]
@orianweste - I do not unfortunately.  Its driving me nuts too.  Very little info on google about it.  The few articles I found were referencing that the disconnect issue was related to trying to reach the google apis via the Firebase SDK.   I managed to fix a Fabric issue (Firebase Crashlytics linking problem) and after that it seems I am not getting too many more disconnect/hanging issues - in fact, I am not certain its happened again since the Fabric fix.  I outlined the Fabric issue and fix here:  https://github.com/dpa99c/cordova-plugin-firebasex/issues/91 

I still get plenty of `BoringSSL nw_protocol_boringssl_get_out_frames` errors though.  Its not crashing the app but they do happen in batches of 4 or 5 about every 20min if I just leave my app running.
